R/rules.R

```diff
@@ -379,3 +413,298 @@ prepare_values_text <- function(set,
+    allow_na = allow_na,
+    allow_nan = allow_nan,
+    allow_inf = allow_inf,
+    operator = "<="
+  )
+}
+
+#' Validate that a field is equal to a specified value
+#'
+#' The `sv_equal()` function compares the field value to a specified value with
+#' the `==` operator.
+#' 
+#' @param rhs The right hand side (RHS) value is to be used for the comparison
+#'   with the field value. The validation check will effectively be of the form
+#'   `<field> == <rhs>`.
+#' @inheritParams sv_gt
+#'
+#' @return A function suitable for using as an
+#'   [`InputValidator$add_rule()`][InputValidator] rule.
+#'
+#' @export
+sv_equal <- function(rhs,
+                     message = "Must be equal to {rhs}.",
+                     allow_multiple = FALSE,
+                     allow_na = FALSE,
+                     allow_nan = FALSE,
+                     allow_inf = FALSE) {
+  
+  force(rhs)
+  force(message)
+  force(allow_multiple)
+  force(allow_na)
+  force(allow_nan)
+  force(allow_inf)
+  
+  # Preparation of the message and the validation test
+  sv_comparison(
+    rhs = rhs,
+    message = message,
+    allow_multiple = allow_multiple,
+    allow_na = allow_na,
+    allow_nan = allow_nan,
+    allow_inf = allow_inf,
+    operator = "=="
+  )
+}
+
+#' Validate that a field is not equal to a specified value
+#'
+#' The `sv_not_equal()` function compares the field value to a specified value
+#' with the `!=` operator.
+#' 
+#' @param rhs The right hand side (RHS) value is to be used for the comparison
+#'   with the field value. The validation check will effectively be of the form
+#'   `<field> != <rhs>`.
+#' @inheritParams sv_gt
+#'
+#' @return A function suitable for using as an
+#'   [`InputValidator$add_rule()`][InputValidator] rule.
+#'
+#' @export
+sv_not_equal <- function(rhs,
+                         message = "Must not be equal to {rhs}.",
+                         allow_multiple = FALSE,
+                         allow_na = FALSE,
+                         allow_nan = FALSE,
+                         allow_inf = FALSE) {
+  
+  force(rhs)
+  force(message)
+  force(allow_multiple)
+  force(allow_na)
+  force(allow_nan)
+  force(allow_inf)
+  
+  # Preparation of the message and the validation test
+  sv_comparison(
+    rhs = rhs,
+    message = message,
+    allow_multiple = allow_multiple,
+    allow_na = allow_na,
+    allow_nan = allow_nan,
+    allow_inf = allow_inf,
+    operator = "!="
+  )
+}
+
+sv_comparison <- function(rhs,
+                          message,
+                          allow_multiple,
+                          allow_na,
+                          allow_nan,
+                          allow_inf,
+                          operator) {
+  
+  # Preparation of the message
+  message <- 
+    glue::glue_data_safe(
+      list(rhs = rhs),
+      message
+    )
+  
+  # Testing of `value` and validation
+  function(value) {
+    
+    # Validity testing of `value` within set constraints
+    if (!is.numeric(value)) {
```

Comment:
Remove this check; let anything try to use the comparison operator (strings, dates)

Reply:
Now removed.